ui/tests/acceptance/jobs-list-test.js

```diff
@@ -736,26 +736,205 @@ module('Acceptance | jobs list', function (hooks) {
   //   });
   // }
 
-  // module('Pagination', function (hooks) {
-  //   hooks.beforeEach(function () {
-  //     const jobsToCreate = 55;
-  //     for (let i = 0; i < jobsToCreate; i++) {
-  //       server.create('job', {
-  //         namespaceId: 'default',
-  //         resourceSpec: Array(1).fill('M: 256, C: 500'),
-  //         groupAllocCount: 1,
-  //         modifyIndex: i + 1,
-  //         createAllocations: false,
-  //       });
-  //     }
-  //   });
-
-  //   // module('Buttons are appropriately disabled', function () {
-  //     test('when there are no jobs', async function (assert) {
-  //       await JobsList.visit();
-  //       assert.ok(JobsList.prevPage.isDisabled, 'Prev page button is disabled');
-  //       assert.ok(JobsList.nextPage.isDisabled, 'Next page button is disabled');
-  //     });
-  //   // });
-  // });
+  module('Pagination', function (hooks) {
+    module('Buttons are appropriately disabled', function () {
+      test('when there are no jobs', async function (assert) {
+        await JobsList.visit();
+        assert.dom('[data-test-pager="first"]').doesNotExist();
+        assert.dom('[data-test-pager="previous"]').doesNotExist();
+        assert.dom('[data-test-pager="next"]').doesNotExist();
+        assert.dom('[data-test-pager="last"]').doesNotExist();
+        await percySnapshot(assert);
+      });
+      test('when there are fewer jobs than your page size setting', async function (assert) {
+        localStorage.setItem('nomadPageSize', '10');
+        createJobs(server, 5);
+        await JobsList.visit();
+        assert.dom('[data-test-pager="first"]').isDisabled();
+        assert.dom('[data-test-pager="previous"]').isDisabled();
+        assert.dom('[data-test-pager="next"]').isDisabled();
+        assert.dom('[data-test-pager="last"]').isDisabled();
+        await percySnapshot(assert);
+        localStorage.removeItem('nomadPageSize');
+      });
+      test('when you have plenty of jobs', async function (assert) {
+        localStorage.setItem('nomadPageSize', '10');
+        createJobs(server, 25);
+        await JobsList.visit();
+        assert.dom('.job-row').exists({ count: 10 });
+        assert.dom('[data-test-pager="first"]').isDisabled();
+        assert.dom('[data-test-pager="previous"]').isDisabled();
+        assert.dom('[data-test-pager="next"]').isNotDisabled();
+        assert.dom('[data-test-pager="last"]').isNotDisabled();
+        // Clicking next brings me to another full page
+        await click('[data-test-pager="next"]');
+        assert.dom('.job-row').exists({ count: 10 });
+        assert.dom('[data-test-pager="first"]').isNotDisabled();
+        assert.dom('[data-test-pager="previous"]').isNotDisabled();
+        assert.dom('[data-test-pager="next"]').isNotDisabled();
+        assert.dom('[data-test-pager="last"]').isNotDisabled();
+        // clicking next again brings me to the last page, showing jobs 20-25
+        await click('[data-test-pager="next"]');
```

Comment:
these are all "next" - how about adding a "previous" click too? maybe a couple, all the way back to page 1?  maybe even also a "first" and a "last" ?

I think I just like the simplicity of this "plenty of jobs" test, so want to push all the buttons with the concise disabled-or-not assertions.

Reply:
Yeah, one of my back-of-napkin "test for this" cases was actually a next-prev-next loop. Let me add that!